AT_Training/Exercise_1/Program.cs

```
@@ -0,0 +1,80 @@
+﻿using System;
+
+namespace Exercise_1
+{
+    class Program
+    {
+        static void Main(string[] args)           
+        {
+            double X, Y, Z;
+            do
+            {
+                try
+                {
+                Console.Write("X:");
+                    X = Convert.ToDouble(Console.ReadLine());
+                    break;
+                }
+                catch (FormatException)
+                {
+                    Console.WriteLine("Please enter a numeric value");
+                }
+            }
+            while (true);
+
+            do
+            {
+                try
+                {
+                Label1: Console.Write("Y:");
+                    Y = Convert.ToDouble(Console.ReadLine());
+                    if (Y == 0)
+                    {
+                        Console.WriteLine("Value cannot be zero");
+                        goto Label1;
```

If you are already using try catch you could throw an exception without goto :)
but this is could be simplified too